torchgen/api/python.py

```diff
@@ -998,17 +998,25 @@ def returns_named_tuple_pyi(signature: PythonSignature) -> Optional[Tuple[str, s
     namedtuple_name = signature.name
     field_names = namedtuple_fieldnames(signature.returns.returns)
     if field_names:
-        namedtuple_def_lines = [f"class {namedtuple_name}(NamedTuple):"]
+        # These types are structseq objects which act like named NamedTuples, but
+        # the constructor acts like the constructor of tuple. Using typing.NamedTuple
+        # does not allow us to override __init__.
+        namedtuple_def_lines = [f"class {namedtuple_name}(Tuple):"]
```

Comment:
Does this pattern give correct typing for indexed access?

Reply:
Fixed now. I tried the following file
```python
from torch.return_types import frexp
import torch
from torch import Tensor

a = frexp((torch.randn(5), torch.randn(5)))
a.mantissa = torch.rand(5)  # error
m: int = a.mantissa  # error
m2: int = a[0]  # error
p: Tensor = a.mantissa
a.mantissa2  # error
b: Tensor = a[0]
c = a[2]  # error
frexp((torch.randn(5), 2)) # error
```
and it gives 6 errors as expected.

Reply:
`reveal_type(a[0])` gives `Any` for me, this patch fixes it for me:
```diff
diff --git a/torchgen/api/python.py b/torchgen/api/python.py
index 2a33fe604da..7a9859a6667 100644
--- a/torchgen/api/python.py
+++ b/torchgen/api/python.py
@@ -1001,11 +1001,11 @@ def returns_named_tuple_pyi(signature: PythonSignature) -> Optional[Tuple[str, s
         # These types are structseq objects which act like named NamedTuples, but
         # the constructor acts like the constructor of tuple. Using typing.NamedTuple
         # does not allow us to override __init__.
-        namedtuple_def_lines = [f"class {namedtuple_name}(Tuple):"]
+        seq_type = f"Tuple[{', '.join(python_returns)}]"
+        namedtuple_def_lines = [f"class {namedtuple_name}({seq_type}):"]
         namedtuple_def_lines.extend(
             f"    {name}: {typ}" for name, typ in zip(field_names, python_returns)
         )
-        seq_type = f"Tuple[{', '.join(python_returns)}]"
         namedtuple_def_lines.append(
             f"    def __init__(self, sequence: {seq_type}): ..."
         )
```

Reply:
Thanks. Already added in e469f09663f190fe5b7826632642f2a3a4de7276